### PR TITLE
Add section on initialising function & delegates

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Comment by andralex** [comment]
Reworded things a bit. Among the changes... https://askubuntu.com/questions/195073/synaptic-initialize-initialise-why-the-difference

**Comment by thewilsonator** [comment]
Thanks!

**Comment by thewilsonator** [comment]
> https://askubuntu.com/questions/195073/synaptic-initialize-initialise-why-the-difference

I'm an Aussie, you can't blame me. Thats habit.

File: spec/function.dd [hunk 1]
```diff
@@ -589,6 +589,25 @@ $(H2 $(LNAME2 optional-parenthesis, Optional Parentheses))
...
+    $(SPEC_RUNNABLE_EXAMPLE_COMPILE
+    ---
+    int function() foo = { return 42; };
+    int delegate() bar = { return 43; };
+    int delegate() baz;
+    void main()
+    {
+        assert(foo() == 42);
+        assert(bar() == 43);
+        assert(bas is null);
```
[the remark this answers is not in the document]
Comment (thewilsonator) [reply spec/function.dd]: Good catch.

**Comment by JinShil** [comment]
DAutoTest Failure:

> Searching for trailing whitespace
./spec/function.dd:592:    
./spec/function.dd:597:

**Comment by thewilsonator** [comment]
Oh I didn't realise there was a sub-section heading that this came under.

**Comment by ZombineDev** [comment]
IIRC, this was meant to be in section 19.19. Ping me when you make the changes and I'll merge.

**Comment by thewilsonator** [comment]
@ZombineDev done. I don't know what is was thinking.

**Pull request merged**